### PR TITLE
PWN-8206 - added at sign to username

### PR DESCRIPTION
**Pull request by eduardmaximovich**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8206

## Description of Work

- added `@` sign to username

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

**Comment by gslevinkov** [comment]
We have special feature toggle for username domain. I think we need to connect it somehow with your method. How about a class UsernameFormatter or smth like that?

**Review by Davran331122 — APPROVED** [review]

**Comment by eduardmaximovich** [comment]
@gslevinkov check please again, I've added UsernameFormatter

File: app/src/main/java/org/p2p/wallet/utils/UsernameFormatter.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        return if (username.endsWith(usernameDomainFeatureToggle.value)) {
+            "@$username"
+        } else {
+            username
+        }
+    }
+
+    fun formatOrNull(username: String?): String? {
+        if (username == null) return null
+        return format(username)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/utils/UsernameFormatter.kt:18]: return username?.let(::format)

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/newsend/repository/RecipientsDatabaseRepository.kt [hunk 2]
```diff
@@ -4,17 +4,19 @@ import org.p2p.wallet.newsend.db.RecipientsDao
 import org.p2p.wallet.newsend.model.RecipientConverter
 import org.p2p.wallet.newsend.model.SearchResult
 import java.util.Date
+import org.p2p.wallet.utils.UsernameFormatter
 
 class RecipientsDatabaseRepository(
-    private val recipientsDao: RecipientsDao
+    private val recipientsDao: RecipientsDao,
+    private val usernameFormatter: UsernameFormatter
 ) : RecipientsLocalRepository {
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/newsend/repository/RecipientsDatabaseRepository.kt:12]: I'm not sure about formatter inside repository, but moving it elsewhere would require more refactoring

**Pull request merged**